electron-app/src/server/websites/cohost/cohost.service.ts

```diff
@@ -0,0 +1,305 @@
+import { Injectable } from '@nestjs/common';
+import {
+  DefaultOptions,
+  FileRecord,
+  FileSubmission,
+  FileSubmissionType,
+  CohostFileOptions,
+  CohostNotificationOptions,
+  PostResponse,
+  Submission,
+  SubmissionPart,
+  SubmissionRating,
+} from 'postybirb-commons';
+import UserAccountEntity from 'src/server/account/models/user-account.entity';
+import { PlaintextParser } from 'src/server/description-parsing/plaintext/plaintext.parser';
+import Http from 'src/server/http/http.util';
+import { CancellationToken } from 'src/server/submission/post/cancellation/cancellation-token';
+import {
+  FilePostData,
+  PostFile,
+  PostFileRecord,
+} from 'src/server/submission/post/interfaces/file-post-data.interface';
+import { PostData } from 'src/server/submission/post/interfaces/post-data.interface';
+import { ValidationParts } from 'src/server/submission/validator/interfaces/validation-parts.interface';
+import FileSize from 'src/server/utils/filesize.util';
+import FormContent from 'src/server/utils/form-content.util';
+import WebsiteValidator from 'src/server/utils/website-validator.util';
+import ImageManipulator from 'src/server/file-manipulation/manipulators/image.manipulator';
+import { LoginResponse } from '../interfaces/login-response.interface';
+import { ScalingOptions } from '../interfaces/scaling-options.interface';
+import { Website } from '../website.base';
+import _ from 'lodash';
+import cheerio from 'cheerio';
+import { CohostFileOptionsEntity } from 'postybirb-commons/lib/websites/cohost/cohost.file.options';
+
+@Injectable()
+export class Cohost extends Website {
+  BASE_URL: string = 'https://cohost.org';
+  readonly MAX_CHARS: number = 5000;
+  acceptsFiles: string[] = ['jpeg', 'jpg', 'jfif', 'png', 'pjpeg', 'pjp', 'gif', 'webp', 'svg', 'aac', 'm4a', 'm4b', 'flac', 'mp3', 'mpega', 'mp2', 'wav'];
+  acceptsAdditionalFiles: boolean = true;
+
+  page_name: string = "";
+  cookie: any;
+  cohostPlus: boolean;
+
+  readonly defaultDescriptionParser = PlaintextParser.parse;
```

Comment:
question: I see in the posting code, the json suggests markdown. If cohost uses markdown it is suggested to use the MarkdownParser.

Reply:
It actually supports HTML directly so I suppose no parser is needed at all?

Reply:
@IsoFrieze It's likely that they don't support HTML directly, but rather allow certain tags through, which many Markdown implementations allow. Technically, the original version allowed ANY HTML tag, which is unsafe. Suggestion: Try inserting some <script> tags. if they work, please contact cohost and let them know they have a major security vulnerability. I would expect these to be stripped out, though.

Reply:
Yeah, apparently they have this page ([https://cohost.org/rc/content/markdown-reference](https://cohost.org/rc/content/markdown-reference)) that explains what's allowed and what's not. It doesn't mention <script> actually, but it definitely gets filtered out of the post entirely.

In the update I did, I just let all the HTML get passed to cohost and then it takes care of stripping all the bad stuff. If there's a better way to go around it, let me know. Otherwise this PR should be ready to go I think.